aws/main.tf

```diff
@@ -0,0 +1,24 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+  }
+
+  required_version = ">= 1.2.0"
+}
+provider "aws" {
+    access_key = "${var.access_key}"
+    secret_key = "${var.secret_key}"
+    region = "us-east-1"
+}
+
+resource "aws_instance" "ec2_instance" {
+    ami = "${var.ami_id}"
```

Comment:
Why are you using var here? Why not just put the values in the main? Is there some reason to have extra files?